### DISCUSSION TITLE
Where does the data come from?

### DISCUSSION THREAD

**Discussion by busaku** [teslamate-org/teslamate#1438] [Q&A]
(Sorry for my englisch)

i love the project. that's how i got interested in the technical side.

where does teslamate get all of the very detailed data from?
Does the car always transfer this to tesla and teslamate reads it out?

I tried to search a little in the source text (but I can't do an elixir). it seems to be a away socket. if I understand that correctly, my car will establish a socket connection to tesla, which teslamate will read (?).

I would be very interested in how exactly this works;)

thanks
pad

**Comment by ngardiner** [comment]
The requests are basically proxied from TeslaMate via Tesla's API to the vehicle. There's no direct connectivity to the vehicle so to speak, but the data comes directly from the vehicle via the API that TeslaMate queries.

**Comment by rmyadsk** [comment]
Yes, the Tesla API (application programming interface) is well-documented [here](https://tesla-api.timdorr.com/).

So, yes, there is a "mothership" of sorts that Teslamate and other third party apps must connect through.

So, if you have a Tesla, your Tesla has a cellular data connection. Over that connection the car will upload data to Tesla. If you are its owner, and if you intend to retrieve your car's data, you can generate one or more tokens to prove that you are you, then you feed that token to the Tesla mothership over and over again via a web service. Fun, right?

Part of the beauty of Teslamate is seamless API integration: you enter your Tesla username and oassword pair once, and it manages tokens for you as needed.

**Comment by busaku** [comment]
Wow, so tesla store a loooot of data of all cars.
Doenst understand why the original app did not show these information.

i love it, thanks :)
